addons/mail/static/src/models/thread/thread.js

```diff
@@ -659,9 +659,12 @@ function factory(dependencies) {
          * Marks as read all needaction messages with this thread as origin.
          */
         async markNeedactionMessagesAsOriginThreadAsRead() {
-            await this.async(() =>
-                this.env.models['mail.message'].markAsRead(this.needactionMessagesAsOriginThread)
-            );
+            await this.async(() => {
+                this.env.models['mail.message'].markAllAsRead([
+                    ['res_id', '=', this.id],
+                    ['model', '=', this.model]
+                ]);
+            });
```

Comment:
I feel like this should actually be linked to thread cache domain then, which is giving the domain we need on each case.

It also makes sense: in the rare case while viewing a filtered thread, this will mark as read only the messages that have been actually viewed regarding the current filter.

So "move" (duplicate in stable) the method in thread cache and take its domain, here call it on the main thread cache (for stable reasons) and also move the compute calling this method on the thread cache itself?

Note the deprecated `markNeedactionMessagesAsRead` should probably have the fix too.

Reply:
Note sure though this is a suggestion. @alexkuhn what do you think?

Reply:
It's wrong for `markNeedactionMessagesAsOriginThreadAsRead` to affect all messages that have given res_id/model.

Rename the method so that it makes sense with what it actually does.

Reply:
I don't understand this comment, isn't origin thread specifically res_id+model?

Reply:
> I don't understand this comment, isn't origin thread specifically res_id+model?

I'm referring to the name of method `markNeedactionMessagesAsOriginThreadAsRead` being misleading with what it actually does. The part with `res_id` and `model` use in `markAllAsRead` maps to origin thread, so that's alright... Not fond of using these raw data rather than thread, but I didn't want to raise this issue because this is more code cleaning and it's not really fitting for fix in stable.

Reply:
@alexkuhn @seb-odoo Well we have to choose something : 

1. Rename
2. Don't rename
3. Rename + duplicate

Personally I would like to avoid the duplication.

Reply:
1. **Rename:** Looks good
2. **Don't rename:** Not good because naming is misleading, and it's not good to have this in code (even if just in stable)
3. **Rename + duplicate:** Don't like duplication neither, because dead code for basically sake of keeping dead code... Again could be misleading without something to let us know it's deprecated... Could be alright if we use `@deprecated` tag I guess.

I would opt for 1 (**Rename**) because the "safe" approach of 3 (**Rename + duplicate**) is not worth it IMO.